src/app/static/src/app/store/language/actions.ts

```diff
@@ -3,8 +3,39 @@ import i18next from "i18next";
 import {LanguageMutation} from "./mutations";
 import {Language} from "../translations/locales";
 import {TranslatableState} from "../../types";
+import {DataExplorationState} from "../dataExploration/dataExploration";
 
 export async function changeLanguage<T extends TranslatableState>({commit}: ActionContext<T, T>, lang: Language) {
     await i18next.changeLanguage(lang);
     commit({type: LanguageMutation.ChangeLanguage, payload: lang})
 }
+
+export const ChangeLanguageAction = async (context: ActionContext<any, any>, payload: Language, root = false) => {
```

Comment:
There's actually a nice way to do this in Typescript without losing typings and without using a boolean argument here:

```
function isRoot(object: DataExplorationState): object is RootState {
    return !object.dataExplorationMode
}

export const ChangeLanguageAction = async (context: ActionContext<DataExplorationState, DataExplorationState>,
                                           payload: Language) => {

    const {commit, dispatch, rootState} = context;

    if (rootState.language === payload) {
        return;
    }

    commit({type: LanguageMutation.SetUpdatingLanguage, payload: true});

    await i18next.changeLanguage(payload);
    commit({type: LanguageMutation.ChangeLanguage, payload: payload})

    const actions: Promise<unknown>[] = [];

    if (rootState.baseline?.iso3) {
        actions.push(dispatch("metadata/getPlottingMetadata", rootState.baseline.iso3));
    }

    if (isRoot(rootState) && rootState.modelCalibrate.status.done) {
        actions.push(dispatch("modelCalibrate/getResult"));
    }

    if (Object.keys(rootState.genericChart.datasets).length > 0) {
        actions.push(dispatch("genericChart/refreshDatasets"));
    }

    await Promise.all(actions);
    commit({type: LanguageMutation.SetUpdatingLanguage, payload: false});
}
```

Reply:
> There's actually a nice way to do this in Typescript without losing typings and without using a boolean argument here:
> 
> ```
> function isRoot(object: DataExplorationState): object is RootState {
>     return !object.dataExplorationMode
> }
> 
> export const ChangeLanguageAction = async (context: ActionContext<DataExplorationState, DataExplorationState>,
>                                            payload: Language) => {
> 
>     const {commit, dispatch, rootState} = context;
> 
>     if (rootState.language === payload) {
>         return;
>     }
> 
>     commit({type: LanguageMutation.SetUpdatingLanguage, payload: true});
> 
>     await changeLanguage<DataExplorationState>(context, payload);
> 
>     const actions: Promise<unknown>[] = [];
> 
>     if (rootState.baseline?.iso3) {
>         actions.push(dispatch("metadata/getPlottingMetadata", rootState.baseline.iso3));
>     }
> 
>     if (isRoot(rootState) && rootState.modelCalibrate.status.done) {
>         actions.push(dispatch("modelCalibrate/getResult"));
>     }
> 
>     if (Object.keys(rootState.genericChart.datasets).length > 0) {
>         actions.push(dispatch("genericChart/refreshDatasets"));
>     }
> 
>     await Promise.all(actions);
>     commit({type: LanguageMutation.SetUpdatingLanguage, payload: false});
> }
> ```

Never seen that used before, really cool.